libcontainer/specconv/spec_linux.go

```diff
@@ -190,6 +195,17 @@ func CreateLibcontainerConfig(opts *CreateOpts) (*configs.Config, error) {
 		if config.Namespaces.Contains(t) {
 			return nil, fmt.Errorf("malformed spec file: duplicated ns %q", ns)
 		}
+		// if this container is going to have shared cgroup namespace with host,
+		// we just skip the following .Add()
```

Comment:
I have found that `docker` just uses `delNamespace()` to delete a namespace shared with host. If a user runs a container without adding cgroup ns in `config.json`, then this container will share cgroup ns with host and its cgroups are bind-mounted. I think it is the same with the scenario that we setup cgroup ns in `config.json` but set cgroup path as `/proc/1/ns/cgroup`. So I chose not to add cgroup ns into `config.Namespaces`. Then its cgroups will be bind-mounted without adding more codes.